xbmc/video/VideoThumbLoader.cpp

```diff
@@ -472,35 +470,28 @@ bool CVideoThumbLoader::FillLibraryArt(CFileItem &item)
       // For episodes and seasons, we want to set fanart for that of the show
       if (!item.HasArt("tvshow.fanart") && tag.m_iIdShow >= 0)
       {
-        ArtCache::const_iterator i = m_showArt.find(tag.m_iIdShow);
-        if (i == m_showArt.end())
+        std::map<std::string, std::string> artmap;
+        if (GetArtFromCache(MediaTypeTvShow, tag.m_iIdShow, artmap))
         {
-          std::map<std::string, std::string> showArt;
-          m_videoDatabase->GetArtForItem(tag.m_iIdShow, MediaTypeTvShow, showArt);
-          i = m_showArt.insert(std::make_pair(tag.m_iIdShow, showArt)).first;
-        }
-        if (i != m_showArt.end())
-        {
-          item.AppendArt(i->second, "tvshow");
+          item.AppendArt(artmap, MediaTypeTvShow);
           item.SetArtFallback("fanart", "tvshow.fanart");
           item.SetArtFallback("tvshow.thumb", "tvshow.poster");
         }
       }
 
       if (!item.HasArt("season.poster") && tag.m_iSeason > -1)
       {
-        ArtCache::const_iterator i = m_seasonArt.find(tag.m_iIdSeason);
-        if (i == m_seasonArt.end())
-        {
-          std::map<std::string, std::string> seasonArt;
-          m_videoDatabase->GetArtForItem(tag.m_iIdSeason, MediaTypeSeason, seasonArt);
-          i = m_seasonArt.insert(std::make_pair(tag.m_iIdSeason, seasonArt)).first;
-        }
-
-        if (i != m_seasonArt.end())
-          item.AppendArt(i->second, MediaTypeSeason);
+        std::map<std::string, std::string> artmap;
+        if (GetArtFromCache(MediaTypeSeason, tag.m_iIdSeason, artmap))
+          item.AppendArt(artmap, MediaTypeSeason);
```

Comment:
`AppendArt` takes `artmap` by reference and allocates a new artmap. It would be more efficient to take advantage of copy elision, return artmap from `GetArtFromCache` by value, then use `std::move` to pass an rvalue to `AppendArt`. That way, the resources we allocated in this function get passed to the item without a second allocation.

Reply:
Sorry, I didn't see where `CGUIListItem::AppendArt` allocates a new map, but the update should avoid creating a map that is often replaced from `GetArtFromCache`.

Reply:
OK, this is fine